src/settings/app_settings.rs

```diff
@@ -1,18 +1,37 @@
 use serde_aux::field_attributes::deserialize_number_from_string;
 
-#[derive(serde::Deserialize, Clone)]
+#[derive(serde::Deserialize, serde::Serialize, Clone)]
 /// Structure representing the application settings
 pub struct ApplicationSettings {
     /// The IP address or the TCP/IP hostname of the SNMP Simulator HTTP server
     /// serving the management REST API.
+    #[serde(default = "default_host")]
     pub host: String,
     #[serde(deserialize_with = "deserialize_number_from_string")]
     /// Specifies the port for which the service is configured to accept client
     /// requests. The port value is used in conjunction with the host name.
+    #[serde(default = "default_port")]
     pub port: u16,
-    #[serde(default)]
+    #[serde(default = "default_uri_prefix")]
     pub uri_prefix: String,
     /// Verbosity level of the logger. Following values are supported error,
     /// warn, info, debug and trace.
+    #[serde(default = "default_verbosity_level")]
     pub level: String,
 }
+
+fn default_host() -> String {
+    "0.0.0.0".to_string()
+}
+
+fn default_port() -> u16 {
+    8080
+}
+
+fn default_uri_prefix() -> String {
+    "mngmt/v1".to_string()
```

Comment:
We can keep the URI empty as default.

FYI: I've just realized that this was a bug in the original base.yaml configuration => the URI must start with a slash '/'